extensions/vscode/snippets/bloc.json

```diff
@@ -74,96 +150,91 @@
       ")"
     ]
   },
-  "BlocProvider": {
-    "prefix": "blocprovider",
-    "body": [
-      "BlocProvider(",
-      "\tcreate: (context) => ${1:Subject}${2|Bloc,Cubit|}(),",
-      "\tchild: ${3:Container()},",
-      ")"
-    ]
+  "BlocProvider.of()": {
+    "prefix": "blocof",
+    "body": "BlocProvider.of<${1:Subject}${2|Bloc,Cubit|}>(context)"
   },
-  "MultiBlocProvider": {
-    "prefix": "multiblocprovider",
-    "body": [
-      "MultiBlocProvider(",
-      "\tproviders: [",
-      "\t\tBlocProvider(",
-      "\t\t\tcreate: (context) => ${1:Subject}${2|Bloc,Cubit|}(),",
-      "\t\t),",
-      "\t\tBlocProvider(",
-      "\t\t\tcreate: (context) => ${3:Subject}${4|Bloc,Cubit|}(),",
-      "\t\t),",
-      "\t],",
-      "\tchild: ${5:Container()},",
-      ")"
-    ]
+  "RepositoryProvider.of()": {
+    "prefix": "repositoryof",
+    "body": "RepositoryProvider.of<${1:Subject}Repository>(context)"
   },
-  "RepositoryProvider": {
-    "prefix": "repositoryprovider",
-    "body": [
-      "RepositoryProvider(",
-      "\tcreate: (context) => ${1:Subject}Repository(),",
-      "\tchild: ${2:Container()},",
-      ")"
-    ]
+  "context.bloc()": {
+    "prefix": "bloc",
+    "body": "context.bloc<${1:Subject}${2|Bloc,Cubit|}>()"
   },
-  "MultiRepositoryProvider": {
-    "prefix": "multirepositoryprovider",
-    "body": [
-      "MultiRepositoryProvider(",
-      "\tproviders: [",
-      "\t\tRepositoryProvider(",
-      "\t\t\tcreate: (context) => ${1:Subject}Repository(),",
-      "\t\t),",
-      "\t\tRepositoryProvider(",
-      "\t\t\tcreate: (context) => ${2:Subject}Repository(),",
-      "\t\t),",
-      "\t],",
-      "\tchild: ${3:Container()},",
-      ")"
-    ]
+  "context.repository()": {
+    "prefix": "repository",
+    "body": "context.repository<${1:Subject}Repository>()"
   },
-  "BlocObserver": {
-    "prefix": "blocobserver",
+  "Bloc state": {
+    "prefix": "blocstate",
     "body": [
-      "import 'package:bloc/bloc.dart';",
+      "class ${1:Subject}${2:Verb}${3:State} extends $1State {",
+      "\tconst $1$2$3($5);",
       "",
-      "class ${1:My}BlocObserver extends BlocObserver {",
-      "\t@override",
-      "\tvoid onEvent(Bloc bloc, Object event) {",
-      "\t\t${2:// TODO: implement onEvent}",
-      "\t\tsuper.onEvent(bloc, event);",
-      "\t}",
+      "$4",
       "",
       "\t@override",
-      "\tvoid onError(Cubit cubit, Object error, StackTrace stacktrace) {",
-      "\t\t${3:// TODO: implement onError}",
-      "\t\tsuper.onError(cubit, error, stacktrace);",
-      "\t}",
+      "\tList<Object> get props => [$6];",
```

Comment:
I don't think we should default to Equatable especially since there is technically also freezed support. Thoughts?

Reply:
I think, more often than not, you do want to use Equatable. Especially when using the initial, in progress, success and failure states. This general state snippet is based on that.

What would you suggest?

Reply:
Yeah thinking about it some more and I'm okay with defaulting to Equatable for now but I don't think we should include the initial, in progress, success, and failure states because we already have the `blocevent` and `blocstate` snippets and they are a bit too opinionated/specific imo. I also kept the `bloc` and `cubit` prefixes but renamed `contextbloc` to `ctxbloc` etc... so hopefully it's a good compromise because it's clear and more concise.

Reply:
@felangel, the initial, in progress, success and failure states (as well as the started event) are entirely based on the naming conventions in the docs. Why do you think they're still too opinionated?

Also, I don't really like `ctxbloc`, since it's not something you're used to typing. I'd rather suggest something like `cbloc` or reverting to `contextbloc`. Thoughts?